lib/PngQuant.js

```diff
@@ -22,6 +22,11 @@ function PngQuant(pngQuantArgs) {
 util.inherits(PngQuant, Stream);
 
 PngQuant.getBinaryPath = memoizeAsync(function (cb) {
+    if(this.binaryPath !== undefined) {
+        cb(this.binaryPath);
```

Comment:
You seem to be [releasing Zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony) here by calling the callback immediately. Also, it should be `cb(null, this.binaryPath);` so `this.binaryPath` isn't interpreted as an error.